sklearn/feature_extraction/text.py

```diff
@@ -852,7 +852,7 @@ def _sort_features(self, X, vocabulary):
         Returns a reordered matrix and modifies the vocabulary in place
         """
         sorted_features = sorted(six.iteritems(vocabulary))
-        map_index = np.empty(len(sorted_features), dtype=np.int32)
+        map_index = np.empty(len(sorted_features), dtype=X.indices.dtype)
```

Comment:
Indices should be of dtype `np.intp`

Reply:
I don't have enough background here - could you link to the PR that added 64-bit indexing to scipy?

At any rate, it seems to be that `indices_dtype` should always be `np.intp` - if that's not possible, then you should pick between `np.int32` (on old scipy) and `np.intp` (on fixed scipy)

Reply:
The indices can be either int32 or int64, scipy.sparse works with both.